services/search-repository.js

```diff
@@ -0,0 +1,66 @@
+const Search = require("../models/Search");
+
+addSearch = async ({ user, game, limit }) => {
+    const search = new Search({
+        user,
+        game,
+        limit,
+        party: [
+            {
+                user,
+            },
+        ],
+    });
+    await search.save();
+
+    return search;
+};
+
+deleteSearch = async ({ searchID }) => {
+    const search = await Search.findByIdAndDelete(searchID);
+    if (!search) {
+        throw new Error("NOT_FOUND");
+    }
+
+    return search;
+};
+
+joinSearch = async ({ searchID, playerID }) => {
+    const search = await Search.findById(searchID);
+    if (!search) {
+        throw new Error("NOT_FOUND");
+    }
+    if (search.party.some((player) => player.user === playerID)) {
+        throw new Error("ALREADY_EXISTS");
+    }
+    search.party.push({ user: playerID });
+    await search.save();
+    return search;
+};
+
+leaveSearch = async ({ searchID, playerID }) => {
+    const search = await Search.findById(searchID);
+    if (!search) {
+        throw new Error("NOT_FOUND");
+    }
+
+    if (!search.party.some((player) => player.user === playerID)) {
+        throw new Error("PLAYER_NOT_FOUND");
+    }
+
+    search.party.splice(
+        search.party.findIndex((player) => {

```

Comment:
It would be more readable if you initialize a variable for `findIndex` on top of splice.

Actually you can improve this block of code and the if above it by:

```
const indexInParty = find index code
if (indexInParty === -1) {
  throw new Error("PLAYER_NOT_FOUND");
}

search.party.splice(indexInParty, 1)
```

`findIndex` can act like `some`. The difference is that if an element exists in the array, `some` will return `true` but `findIndex` will return a non-negative integer which represents the index of the first matching item found in the array.

If an element does not exist, then `some` will return `false` and `findIndex` will return `-1` to signify that the item is not found. See [the return value documentation for `findIndex` for more info](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex).